components/login-form.tsx

```
@@ -40,13 +41,51 @@ export function LoginForm() {
+
+  const oauth_services = ['github','google']
+  type OAuthService = typeof oauth_services[number]
+
+  async function handleOauth(oauth_type: OAuthService){
+    try{
+      const response = await exisitingGithubUser();
+      if(!response?.success){
+        const res = await OauthLogin(oauth_type);
+        const auth_url = res?.message.url 
+        window.open(auth_url, "_blank", "width=500,height=600");
+        useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)